### ISSUE TITLE
match/when returning values

### ISSUE THREAD

**Issue by lemunozm** [nlfiedler/oxidized#5]
Hi!

First of all, I like it a lot your library, I feel close to my dear rust :)

I played with it, and I have missed the possibility to return a value from `match`/`when` functions (similar behaviour found in `fold` from `dartz`). Maybe a signature like this is more suitable for more circumstances:

```
R match<R>(R Function(T) some, R Function() none);
R when<R>({required R Function(T) some, required R Function() none});
```
I think it could be easy to implement. I could make a PR if you want.

Thanks!

**Comment by nlfiedler** [comment]
Wow, they are literally the only two functions that return nothing, I hadn't even noticed that. It seems like I never  used the functions myself so I didn't notice the problem. Certainly, I would welcome a PR to fix this, thank you for pointing it out, and I'm glad you like the library.

**Comment by nlfiedler** [comment]
Fixed in commit 732a43e1b601389b19224de0f18ae34cc87e76a2
